spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/functions/IcebergVersionFunctionImpl.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.functions;
+
+import org.apache.iceberg.IcebergBuild;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.catalog.functions.ScalarFunction;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.unsafe.types.UTF8String;
+
+/**
+ * {@link org.apache.spark.sql.connector.catalog.functions.BoundFunction} that implements {@link
+ * org.apache.iceberg.spark.functions.IcebergVersionFunction} by returning the short version string,
+ * e.g. "0.14.0" or "0.15.0-SNAPSHOT".
+ */
+public class IcebergVersionFunctionImpl implements ScalarFunction<UTF8String> {
+
+  private static final String icebergVersionAsString = IcebergBuild.version();
```

Comment:
I don't think that this is needed. It's only used to create the Spark string, and it is available from `IcebergBuild`. Plus, this isn't converting the value to another representation so I don't think you need an "as string" form.

Reply:
Yeah removed.